.github/workflows/haskell.yml

```
@@ -152,3 +155,25 @@ jobs:
     - uses: actions/checkout@v3
     - name: Format all cabal files
       run: ./scripts/cabal-format.sh
+
+  gen-hie:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+
+    steps:
```

Adding caching could speed things up:
```suggestion
    steps:
     - uses: actions/cache@v3 
       name: Cache cabal store 
       with: 
         path: | 
           ${{ steps.setup-haskell.outputs.cabal-store }} 
           dist-newstyle

```

See example here: https://github.com/input-output-hk/cardano-ledger/blob/61bcb797babc992cf079f0333d870720e83f2d10/.github/workflows/haskell.yml#L89-L95

---

Awesome! Super thanks for this suggestion!